Cython/Compiler/ParseTreeTransforms.py

```diff
@@ -3114,6 +3115,21 @@ def _inject_eval(self, node, func_name):
                     node.pos, self.current_scope_node(), lenv))
         return node
 
+    def _inject_class(self, node):
+        if node.name == u'__class__':
```

Comment:
Since this covers the complete function body (and the function name suggests that it does what it says), I'd move this `if` into `visit_NameNode()` above.